labs/aws/go/lab-04/README.md

```diff
@@ -0,0 +1,389 @@
+# Deploying Containers to a Kubernetes Cluster
+
+In this lab, you will deploy a containerized application to a Kubernetes cluster.
+
+> This lab assumes you have a project set up. If you don't yet, please [complete this lab first](../01-iac/01-creating-a-new-project.md).
+
+## Step 1 &mdash; Creating a Kubernetes Cluster
+
+> If you do not have an EKS cluster, you can create one by using the code [here](./code/step1.go).
+
+Point the `KUBECONFIG` environment variable at your cluster configuration file:
+
+```bash
+export KUBECONFIG=~/iac-workshop/kubeconfig
+```
+
+To test out connectivity, run `kubectl cluster-info`. You should see information similar to this:
+
+```
+Kubernetes master is running at https://abcxyz123.gr7.eu-central-1.eks.amazonaws.com
+CoreDNS is running at https://abcxyz123.gr7.eu-central-1.eks.amazonaws.com/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+```
+
+## Step 2 &mdash; Add a func that generates the Kubeconfig
+
+In your project, add the following function:
+
+```go
+//Create the KubeConfig Structure as per https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html
+func generateKubeconfig(clusterEndpoint pulumi.StringOutput, certData pulumi.StringOutput, clusterName pulumi.StringOutput) pulumi.StringOutput {
```

Comment:
Step 1 refers to a kubeconfig from a new cluster. Should we consider moving this from step 2 into step1?